src/targets/rtr.rs

```diff
@@ -52,20 +52,32 @@ impl Tcp {
     }
 
     /// Runs the target.
-    pub async fn run(mut self, component: Component) -> Result<(), ExitError> {
-        let mut notify = NotifySender::new();
+    pub async fn run(self, component: Component) -> Result<(), ExitError> {
+        let notify = NotifySender::new();
         let target = Source::new(self.history_size);
         for &addr in &self.listen {
             self.spawn_listener(addr, target.clone(), notify.clone())?;
         }
 
+        self.run_loop(component, target, notify).await
+    }
+
+    /// Runs the target’s main loop.
+    async fn run_loop(
+        mut self,
+        component: Component,
+        target: Source,
+        mut notify: NotifySender
+    ) -> Result<(), ExitError> {
```

Comment:
How does this code ever break out of the loop or return anything?

Reply:
It doesn’t. This never returns. Maybe it should, but perhaps that is a separate PR?